libretro-rs/src/lib.rs

```diff
@@ -54,19 +67,42 @@ pub trait RetroCore: Sized {
   /// required for the core to function properly.
   fn load_game(env: &mut RetroEnvironment, game: RetroGame) -> RetroLoadGameResult<Self>;
 
-  fn load_game_special(&mut self, env: &mut RetroEnvironment, game_type: u32, info: RetroGame, num_info: usize) -> bool {
-    false
+  fn load_game_special(
+    env: &mut RetroEnvironment,
+    game_type: Self::SpecialGameType,
+    info: RetroGame,
+  ) -> RetroLoadGameResult<Self> {
+    RetroLoadGameResult::Failure
   }
 
   fn unload_game(&mut self, env: &mut RetroEnvironment) {}
 
-  fn get_memory_data<'a>(&'a mut self, env: &mut RetroEnvironment, id: u32) -> Option<&'a mut [u8]> {
+  fn get_memory_data(
+    &mut self,
+    env: &mut RetroEnvironment,
+    _id: RetroMemoryType<Self::SubsystemMemoryType>,
+  ) -> Option<&mut [u8]> {
     None
   }
 
-  fn get_memory_size(&self, env: &mut RetroEnvironment, id: u32) -> usize {
+  fn get_memory_size(&self, _env: &mut RetroEnvironment, _id: RetroMemoryType<Self::SubsystemMemoryType>) -> usize {
```

Comment:
I see that these parameters are unused, so they have `_` prefixes. However, `load_game_special` also has unused parameters that don't have `_` prefixes. Is there a difference between the two that justifies the discrepancy?

Reply:
I think I'm going to remove the underscores. I normally get warnings from CLion/clippy in normal `impl` blocks but since this is part of the trait declaration, the underscore would show up in documentation.